_posts/2018-12-20-so-long-and-thanks.md

```diff
@@ -0,0 +1,131 @@
+and with unavoidable external factors (eg. QA team backlogged, Heroku being down),
+engineers could lose a day or two of productive work. Therefore, the impact of a lost day
+in a two-week sprint compared to a three-week sprint proves to be much more significant.
+* Due to the prior points and the lack of slack time to work on tech debt, the non-immediate
+yet non-trivial tech debt tasks were piling up. Ideally, there was to be a portion of
+time during the sprint dedicated towards tackling technical debt; however, this did not work out
+due to the above reducing the time allotted for prioritized feature work, leaving no room for it.
+
+These are just some of the observations I made during this slightly chaotic time - others
+might have completely different interpretations and thoughts of the matter.
+As of the time of writing this post, we have reverted back to a three-week sprint with a week's
+worth of slack time. From this, I learned that processes can and will change, but they don't necessarily
+need to be permanent and should be critically assessed to see if they work out or not.
+
+# Bite off a Bit More than You Can Chew
+I had absolutely no idea how much I could take on in terms of workload and deliver in a timely manner.
+What if I didn't finish my cards for the sprint? Or what if I break other features with my new one?
+But with time I've come to realize that it wasn't about delivering everything on time or having a perfect
+feature, but rather to extend myself a bit more every time and fail often. During one sprint I had just one card and was hesitant on asking for more (even though they weren't big tasks), but I took another and it went just fine. And although it might not work out everytime, failure in software engineering is the _expected_ outcome, so it's okay. Just as with testing, you'll be constantly disappointed
+if you expect them to work the first time around. 
+
+# `Hash#Dig` then Dig a Bit More... Even If You Find a `nil`
+So during my last few weeks here, I decided to pick up an seemingly straightforward
+3 point task that involved pulling date info from LexisNexis and Experian to compare with
+data from a merchant's application. All I was actually doing was writing a small service
+to grab the data and load it in the ERB template for the view. Little did I know how awful
+APIs could be as well as that it would take me a month to complete this tiny feature.
+I would say that I spent maybe a couple days actually writing the service and helper to create
+the view, but pulling the data was a nightmare. Having never worked with the Underwriting App
+before, diving deep into binary payload and digging out 10-times deep nested hashes for one
+Date object was not the most fun thing I had in mind.
+
+In the beginning I thought it was a one hour task, as the documents had methods that related to 
+the LexisNexis and Experian date objects which I believed to correspond to what I needed.
```

Comment:
Since this is going out into the public, let's not name specific vendors. "Background check vendor" and "credit bureau" would be better options here.